nilearn/plotting/html_report.py

```diff
@@ -111,3 +115,79 @@ def _embed_img(display):
     data = base64.b64encode(io_buffer.read())
 
     return '{}'.format(data.decode())
+
+
+class HTMLReport(plot_utils.HTMLDocument):
+    """ A report written as html
+    """
+    def __init__(self, head_tpl, body):
+        """ The head_tpl is meant for display as a full page, eg writing on
+            disk. The body is used for embedding in an existing page.
+        """
+        self.head_tpl = head_tpl
+        self.body = body
+
+    def _repr_html_(self):
+        """
+        Used by the Jupyter notebook.
+
+        Users normally won't call this method explicitely.
+        """
+        return self.body
+
+    def __str__(self):
+        return self.body
+
+    def get_standalone(self):
+        return self.head_tpl.substitute(body=self.body)
+
+    # save_as_html, open_in_browser are inherited from HTMLDocument
+
+
+class ReportMixin():
+    """
+    Generate a report for Nilearn objects.
+
+    Report is useful to visualize steps in a processing pipeline.
+    Example use case: visualize the overlap of a mask and reference image
+    in NiftiMasker.
+
+    Returns
+    -------
+    report : HTML report with embedded content
+    """
+
+    def update_template(self, title, docstring, content,
```

Comment:
If we pull this out of the ReportMixin class (which is now gone), we could do:

```suggestion
   def update_template(title, docstring, content,
                    parameters, description=None):
    """
    Populate a report with content.

    Parameters
    ----------
    title: str
        The title for the report
    content: img
        The content to display
    description: str
        An optional description of the content

    Returns
    -------
    html : populated HTML report
    """
    body_template_name = 'report_body_template.html'
    body_template_path = os.path.join(
            os.path.dirname(__file__), 'data', 'html', body_template_name)
    tpl = tempita.HTMLTemplate.from_filename(body_template_path,
                                             encoding='utf-8')

    body = tpl.substitute(title=title, content=content,
                          docstring=docstring,
                          parameters=parameters,
                          description=description)
    head_template_name = 'report_head_template.html'
    head_template_path = os.path.join(
            os.path.dirname(__file__), 'data', 'html', head_template_name)
    with open(head_template_path, 'r') as head_file:
        head_tpl = Template(head_file.read())
    return HTMLReport(body=body, head_tpl=head_tpl)
```

Reply:
Actually, it looks like you have that ! :tada: Then we'd just need to pull it out of ReportMixin and it should be perfect. Thank you !!!